cmd/sonobuoy/app/gen_test.go

```diff
@@ -85,19 +91,31 @@ func TestResolveConformanceImage(t *testing.T) {
 func TestResolveConfig(t *testing.T) {
 	defaultPluginSearchPath := config.New().PluginSearchPath
 	defaultAggr := plugin.AggregationConfig{TimeoutSeconds: 10800}
+	dynamicConfigFileName := "*determinedAtRuntime*"
+	defaultConfig := `{"Server":{"bindaddress":"10.0.0.1"}}`
 
 	tcs := []struct {
-		name     string
-		input    *genFlags
+		name string
+
+		// CLI input to parse.
+		input string
+
+		// Not every field of this will be tested at this time.
 		expected *config.Config
-		cliInput string
+
+		// If specified, will write the config to a temp file then append
+		// the `--config=tmpfile` to the input. This way we can keep the config files
+		// small and with the test rather than in a testdata file.
+		configFileContents string
+
+		// TODO(jschnake): This test previously was just testing the config.Config
+		// and only certain fields. We may consider expanding this to testing the entire
+		// genConfig object but for now I am expanding this just to ensure proper plugin loading.
+		expectedGenConfigPlugins *client.GenConfig
 	}{
 		{
-			name: "NonDisruptiveConformance mode when supplied config is nil (nothing interesting happens)",
-			input: &genFlags{
```

Comment:
The test used to be a bit confusing since it could be given a genFlags object or it could be given cmdline input.

Now the test is a bit more consistent by always parsing cmdline input and relying on actual config files.